### PR TITLE
Video listing

### PR DESCRIPTION
**Pull request by AnuragSahu11**
*Added , add to watch later functionality
*Added routes for playlist, watchlater, liked videos, explore videos, history

https://deploy-preview-2--nostalgic-wright-44e896.netlify.app/videoListing/
You can add video to watch later by clicking on the watchlater button on the card, go to watchlater by clicking on the watchlater icon in the sidebar

### CODE REVIEW

**Review by saadahmadhere — APPROVED** [review]
Nice work, clean and readable code.

File: TradePlay/src/components/card/add-to-watchlater.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import { useVideos } from "../../context/videos-context";
+import { inList } from "../../utils/in-list";
+
+const AddToWatchLater = ({ videoData }) => {
+  const { videoState, videoDispatch } = useVideos();
+  const [inWatchlater, setWatchLater] = useState(
+    inList(videoState.watchlater, videoData._id)
+  );
+
+  useEffect(() => {
```
Comment (GuptaEesh) [inline TradePlay/src/components/card/add-to-watchlater.jsx:11]: Is useEffect necessary here? It's normally used for side-effects.
I am not sure if you are handling any side effects here.

**Review by GuptaEesh — COMMENTED** [review]
Good use of array manipulator functions.
Good use of barrel imports.
Just added a suggestion otherwise good to merge

File: TradePlay/src/components/card/add-to-watchlater.jsx (hunk 1 above)
Comment (AnuragSahu11) [reply TradePlay/src/components/card/add-to-watchlater.jsx:11]: Yes I have planned to change that after auth, for now it's temperory. Thanks for feedback.

**Pull request merged**